mediapipe/tasks/ios/vision/image_segmenter/utils/sources/MPPImageSegmenterResult+Helpers.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "mediapipe/tasks/ios/vision/image_segmenter/sources/MPPImageSegmenterResult.h"
+
+#include "mediapipe/framework/packet.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MPPImageSegmenterResult (Helpers)
+
+/**
+ * Creates an `MPPImageSegmenterResult` from confidence masks, category mask and quality scores
+ * packets.
+ *
+ * If `shouldCopyMaskPacketData` is set to `YES`, the confidence and catergory masks of the newly
+ * created `MPPImageSegmenterResult` holds references to deep copied pixel data of the output
+ * respective masks.
+ *
+ * @param confidenceMasksPacket A MediaPipe packet wrapping a `std::vector<mediapipe::Image>`.
+ * @param categoryMaskPacket A MediaPipe packet wrapping a `<mediapipe::Image>`.
+ * @param qualityScoresPacket a MediaPipe packet wrapping a `std::vector<float>`.
+ * @param shouldCopyMaskPacketData A `BOOL` which indicates if the pixel data of the output masks
```

Comment:
Do you think this will be needed?

Reply:
Yeah we have two sets of APIs for each running mode as per Java. Image and video have apis which return a reference or copy of the underlying pixel buffer. So based on which API is invoked, we need to pass this info to MPMask when it is initialised to take appropriate action.